samples/client/petstore/typescript-node/npm/model/models.ts

```diff
@@ -42,7 +42,7 @@ let typeMap: {[index: string]: any} = {
 
 export class ObjectSerializer {
     public static findCorrectType(data: any, expectedType: string) {
-        if (data == undefined) {
+        if (data === undefined) {
```

Comment:
code below assumes `data` is non-null and tries to access its property: 
```
if (data[discriminatorProperty])
```

original code (`== undefined`) would check for both undefined and null. Now `null` remains unchecked, and can lead to NPE at runtime

Reply:
@amakhrov many thanks for the review.

I think the assumption here is rather implicit rather than explicit, so would you agree if I change this PR to `if (data === undefined || data === null)`?

Reply:
yeah that would make sense!